pkg/gadgets/dns/tracer/tracer.go

```diff
@@ -194,10 +288,19 @@ func parseDNSEvent(rawSample []byte) (ret string, pktType string) {
 		pktType = pktTypeNames[pktTypeUint]
 	}
 
+	qTypeUint := uint(dnsEvent.qtype)
+	// Magical values comes from:
+	// https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+	if qTypeUint == 54 || (66 <= qTypeUint && qTypeUint <= 98) || (110 <= qTypeUint && qTypeUint <= 248) || (261 <= qTypeUint && qTypeUint <= 32767) || 32770 <= qTypeUint {
+		qType = "UNASSIGNED"
+	} else {
+		qType = qTypeNames[qTypeUint]
+	}
```

Comment:
What about doing something like:

```
qType, ok := qTypeNames[qTypeUint]
if !ok {
    qType = "UNASSIGNED"
}
```

Reply:
Thank you for it, it is far cleaner than what I wrote (I was too `C`-focused).